### PR TITLE
Add ObjectMetadataAction with PR title and description

### PR DESCRIPTION
**Pull request by jaguilar-atl**
## Overview

Adds an `ObjectMetadataAction` with the PR title and description for heads of`BitbucketPullRequestSCMHead` type.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
                 <artifactId>bom-2.401.x</artifactId>
-                <version>2102.v854b_fec19c92</version>
+                <version>2133.v2e6c00fe4d61</version>
```
Comment (jaguilar-atl) [inline pom.xml:45]: This update is required so we can use the changes introduced in https://github.com/jenkinsci/branch-api-plugin/pull/374 which allow using the display name from the `ObjectMetadataAction` defined for the `SCMHead` associated with the job.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java [hunk 2]
```diff
@@ -140,6 +141,12 @@ protected List<Action> retrieveActions(SCMHead head,
                 .findAny()
                 .ifPresent(action -> result.add(new PrimaryInstanceMetadataAction()));
         }
+
+        if (head instanceof BitbucketPullRequestSCMHead) {
+            MinimalPullRequest pullRequest = ((BitbucketPullRequestSCMHead) head).getPullRequest();
+            result.add(new ObjectMetadataAction(pullRequest.getTitle(), pullRequest.getDescription(), null));
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:147]: Is there any value in providing the `ObjectUrl` as well?

**Review by dkjellin — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (jaguilar-atl) [reply pom.xml:45]: Sorry, I should have been more explicit with my comment. This was so we could utilize the `MultiBranchProjectDisplayNamingTrait` that was added to the `branch-api-plugin` which is one way for users to have the PR title be displayed as the job name.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java (hunk 2 above)
Comment (jaguilar-atl) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java:147]: Thanks for raising. Happy to add it in as well.

File: pom.xml (hunk 1 above)
Comment (jaguilar-atl) [reply pom.xml:45]: We're not referencing it directly ourselves but by upgrading the `branch-api-plugin` to a version that has it, the trait is "automatically" made available as an available option (as shown in one of the screenshots above).

**Comment by jaguilar-atl** [comment]
- [ ] Add unit tests for the new retrieveAction behavior.

File: pom.xml (hunk 1 above)
Comment (jaguilar-atl) [reply pom.xml:45]: Yeah, that's actually a good point. This was mostly to make it more convenient for the user to have the option readily available as part of this feature. However, I've just had a closer look at the bom version itself and it seems that there are some breaking changes included with such an upgrade. With that said, I think it's best to revert this change and as you mentioned the user can always install a newer version of the branch-api-plugin if they require it.

File: src/test/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSourceTest.java [hunk 3]
```diff
@@ -15,8 +18,10 @@
 import hudson.model.Actionable;
 import jenkins.branch.MultiBranchProject;
 import jenkins.scm.api.*;
+import jenkins.scm.api.metadata.ObjectMetadataAction;
 import jenkins.scm.api.metadata.PrimaryInstanceMetadataAction;
 
+import okhttp3.HttpUrl;
```
[the remark this answers is not in the document]
Comment (dkjellin) [reply src/test/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSourceTest.java:24]: We have not yet evaluated that. We have a short backlog of things to look at now that we are Java 11+ only.  We would be happy to compare notes once we get to doing the evaluation (which right now is fairly low priority for us)

**Pull request merged**